### PR TITLE
[CALCITE-5253] Natural join fields validation partially broken after CALCITE-5171

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

**Review by asolimando — CHANGES_REQUESTED** [review]
LGTM in general, the "nit" comments are up to you but please fix the javadoc and try to split up the test into multiple tests

File: core/src/main/java/org/apache/calcite/rel/type/RelCrossType.java
Comment (zstan) [reply core/src/main/java/org/apache/calcite/rel/type/RelCrossType.java:66]: good catch, thanks !

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java
Comment (zstan) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:6167]: fixed, thanks !

**Review by asolimando — APPROVED** [review]
LGTM, thanks for addressing the comments from me and @libenchao.

The Travis failure is due to Travis and unrelated to your patch.

Can you squash the commit messages into a single one?

I will merge within 48 hours if there won't be any objections from others.

**Comment by zstan** [comment]
@asolimando @libenchao guys, thanks for review ! commit messages are squashed.

**Comment by asolimando** [comment]
@zstan, the commit message should match the (current) Jira ticket title, feel free to amend it yourself or I will take care of it before merging.

**Comment by zstan** [comment]
ups, done, thanks.

**Pull request merged**
